hw/ip/otbn/util/shared/information_flow.py

```diff
@@ -143,7 +144,7 @@ def update(self, other: 'InformationFlowGraph') -> None:
             # Updating a nonexistent graph with another graph should return the
             # other graph; since we need to modify self, we change this graph's
             # flow to match other's.
-            self.flow = other.flow.copy()
+            self.flow = deepcopy(other.flow)
```

Comment:
Again, hmmmmm about the deepcopy. Wouldn't it make more sense to just define a sensible `copy` method (instead of `__deepcopy__` below)?